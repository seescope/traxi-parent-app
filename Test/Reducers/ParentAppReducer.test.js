import { TEST_UUID, TEST_REPORTS } from '../Mocks';
import parentAppReducer from '../../App/Reducers/ParentAppReducer';
import {
  RESET_STATE,
  enterKidName,
  selectKidImage,
  selectPrice,
  deviceUpdated,
  addKid,
  selectKid,
} from '../../App/Actions/Actions';
import Intercom from 'react-native-intercom';

it('FETCHED_REPORT', () => {
  const state = { reports: {} };
  const FETCHED_REPORT = { type: 'FETCHED_REPORT', report: TEST_REPORTS, UUID: TEST_UUID };
  const newState = parentAppReducer(state, FETCHED_REPORT);

  expect(newState).toMatchSnapshot();
});

it('RESET_STATE', () => {
  const dirtyState = { kidSuggestions: ['abc123'], selectedKid: 'alice', step: 99 };
  const newState = parentAppReducer(dirtyState, RESET_STATE);

  expect(newState).toMatchSnapshot();
});

it('ENTER_KID_NAME', () => {
  const oldState = { selectedKid: { name: 'Jeff', UUID: '123' } };
  const newState = parentAppReducer(oldState, enterKidName('Bob'));

  expect(newState).toMatchSnapshot();
});

it('SELECT_KID', () => {
  const TEST_KID = { UUID: 'abc123' };
  const oldState = { selectedKid: TEST_KID };
  const newState = parentAppReducer(oldState, selectKid(TEST_KID));

  expect(newState).toMatchSnapshot();
});

it('SELECT_KID_IMAGE', () => {
  const oldState = { selectedKid: {} };
  const newState = parentAppReducer(oldState, selectKidImage('file://something'));

  expect(newState).toMatchSnapshot();
});

it('ADD_KID', () => {
  const oldState = { selectedKid: { UUID: 'nope' }, kids: [] };
  const newState = parentAppReducer(oldState, addKid('new-UUID', 'setupID'));

  expect(newState).toMatchSnapshot();
});

it('DEVICE_UPDATED', () => {
  const oldState = { selectedKid: { UUID: 'nope' }, kids: [], step: 1 };
  const newState = parentAppReducer(
    oldState, deviceUpdated({ UUID: 'surprise', deviceType: 'unknown' })
  );

  expect(newState).toMatchSnapshot();

  const anotherState = parentAppReducer(newState,
    deviceUpdated({ UUID: 'surprise', deviceType: 'Android' })
  );
  expect(anotherState).not.toEqual(newState);
  expect(anotherState).toMatchSnapshot();
});

it('LOGGED_IN', () => {
  const oldState = { profile: { UUID: 'nope' }, kids: [], step: 1, price: 'one dollar' };
  const newState = parentAppReducer(oldState, { type: 'LOGGED_IN', profile: {
    name: 'Jim Bob',
    kids: ['hey', 'ho'],
    UUID: 'orange',
  } });

  expect(newState).toMatchSnapshot();
  expect(Intercom.registerIdentifiedUser).toHaveBeenCalledWith({ userId: 'orange' });
});

it('REACT_NATIVE_ROUTER_FLUX_PUSH', () => {
  const oldState = { sceneName: 'something', other: 'remains' };
  const newState = parentAppReducer(
    oldState, { type: 'REACT_NATIVE_ROUTER_FLUX_PUSH', key: 'hey' }
  );

  expect(newState).toMatchSnapshot();
});

it('PREVIOUS_STEP', () => {
  const oldState = { step: 1 };

  const newState = parentAppReducer(
    oldState, { type: 'PREVIOUS_STEP' }
  );

  expect(newState).toMatchSnapshot();

  const sameState = parentAppReducer(
    newState, { type: 'PREVIOUS_STEP' }
  );
  expect(newState).toEqual(sameState);
});
