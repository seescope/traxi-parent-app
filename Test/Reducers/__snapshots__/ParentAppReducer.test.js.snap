exports[`test ADD_KID 1`] = `
Object {
  "kids": Array [],
  "selectedKid": Object {
    "UUID": "new-UUID",
    "deviceType": "unknown",
    "setupID": "setupID",
    "status": "WAITING",
  },
}
`;

exports[`test DEVICE_UPDATED 1`] = `
Object {
  "kids": Array [],
  "selectedKid": Object {
    "UUID": "surprise",
    "deviceType": "unknown",
  },
  "step": 1,
}
`;

exports[`test DEVICE_UPDATED 2`] = `
Object {
  "kids": Array [],
  "selectedKid": Object {
    "UUID": "surprise",
    "deviceType": "Android",
  },
  "step": 2,
}
`;

exports[`test ENTER_KID_NAME 1`] = `
Object {
  "selectedKid": Object {
    "UUID": "123",
    "name": "Bob",
  },
}
`;

exports[`test FETCHED_REPORT 1`] = `
Object {
  "loading": false,
  "reports": Object {
    "a5f1bc36-89f5-4212-98c6-e92797a2a220": Object {
      "2016-07-08": Object {
        "circles": Array [
          Object {
            "minutesUsed": 0,
            "name": "Late Night",
            "status": "good",
          },
          Object {
            "minutesUsed": 24,
            "name": "Device Time",
            "status": "good",
          },
          Object {
            "minutesUsed": 4,
            "name": "Social Time",
            "status": "good",
          },
        ],
        "trail": Array [
          Object {
            "name": "Evening",
            "trailItems": Array [
              Object {
                "category": "Social",
                "logo": "https://lh3.googleusercontent.com/aYbdIM1abwyVSUZLDKoE0CDZGRhlkpsaPOg9tNnBktUQYsXflwknnOn2Ge1Yr7rImGk=w300",
                "minutesUsed": 2,
                "name": "Instagram",
                "timeStamp": "2016-07-08T17:05:01+10:00",
              },
            ],
          },
          Object {
            "name": "Afternoon",
            "trailItems": Array [
              Object {
                "category": "Travel & Local",
                "logo": "https://lh3.googleusercontent.com/MOf9Kxxkj7GvyZlTZOnUzuYv0JAweEhlxJX6gslQvbvlhLK5_bSTK6duxY2xfbBsj43H=w300",
                "minutesUsed": 2,
                "name": "Google Maps",
                "timeStamp": "2016-07-08T16:58:01+10:00",
              },
              Object {
                "category": "Communication",
                "logo": "https://lh3.googleusercontent.com/nYhPnY2I-e9rpqnid9u9aAODz4C04OycEGxqHG5vxFnA35OGmLMrrUmhM9eaHKJ7liB-=w300",
                "minutesUsed": 6,
                "name": "Google Chrome",
                "timeStamp": "2016-07-08T15:45:01+10:00",
              },
              Object {
                "category": "Social",
                "logo": "https://lh3.googleusercontent.com/aYbdIM1abwyVSUZLDKoE0CDZGRhlkpsaPOg9tNnBktUQYsXflwknnOn2Ge1Yr7rImGk=w300",
                "minutesUsed": 2,
                "name": "Instagram",
                "timeStamp": "2016-07-08T14:01:01+10:00",
              },
            ],
          },
          Object {
            "name": "Morning",
            "trailItems": Array [
              Object {
                "category": "Tools",
                "logo": "https://lh3.googleusercontent.com/DKoidc0T3T1KvYC2stChcX9zwmjKj1pgmg3hXzGBDQXM8RG_7JjgiuS0CLOh8DUa7as=w300",
                "minutesUsed": 6,
                "name": "Google Search",
                "timeStamp": "2016-07-08T11:43:01+10:00",
              },
              Object {
                "category": "Music & Audio",
                "logo": "https://lh3.googleusercontent.com/UrY7BAZ-XfXGpfkeWg0zCCeo-7ras4DCoRalC_WXXWTK9q5b0Iw7B0YQMsVxZaNB7DM=w300",
                "minutesUsed": 6,
                "name": "Spotify",
                "timeStamp": "2016-07-08T11:43:01+10:00",
              },
            ],
          },
        ],
      },
    },
  },
}
`;

exports[`test LOGGED_IN 1`] = `
Object {
  "kids": Array [],
  "parentName": "Jim",
  "price": "one dollar",
  "profile": Object {
    "UUID": "orange",
    "kids": Array [
      "hey",
      "ho",
    ],
    "name": "Jim Bob",
  },
  "step": 1,
}
`;

exports[`test PREVIOUS_STEP 1`] = `
Object {
  "step": 0,
}
`;

exports[`test REACT_NATIVE_ROUTER_FLUX_PUSH 1`] = `
Object {
  "other": "remains",
  "sceneName": "something",
}
`;

exports[`test RESET_STATE 1`] = `
Object {
  "kidSuggestions": Array [],
  "selectedKid": "alice",
  "step": 0,
}
`;

exports[`test SELECT_KID 1`] = `
Object {
  "selectedKid": Object {
    "UUID": "abc123",
    "deviceType": "unknown",
  },
}
`;

exports[`test SELECT_KID_IMAGE 1`] = `
Object {
  "selectedKid": Object {
    "avatarURL": "file://something",
  },
}
`;
